### PR TITLE
Add static security analysis with openstack/bandit, and address initial findings

### PR DESCRIPTION
**Pull request by hangtwenty**
## 🕵️  Add 'bandit' security static analysis to tox & Travis  (71e065e)

Documented with some comments in tox.ini, which should explain I hope!

The official bandit docs are good too.

https://github.com/openstack/bandit

I separated into a 'baseline testing' that is stricter, but more actionable as it can run locally; and 'gatekeeping' that only fails-out on HIGH severity, and that part runs in Travis to put the breaks on and
have us fix or mitigate the issue before merge.

## 🔒 Swap lxml -> defusedxml to fix a security finding (97ed091)

Bandit reported:

    --------------------------------------------------
    >> Issue: [B320:blacklist] Using lxml.etree.parse to parse untrusted XML data is known to be vulnerable to XML attacks. Replace lxml.etree.parse with its defusedxml equivalent function.
       Severity: Medium   Confidence: High
       Location: eemeter/io/parsers.py:462
    461	        try:
    462	            self.root = etree.parse(xml)  # xml is file path or file object
    463	        except IOError:

    --------------------------------------------------
    >> Issue: [B320:blacklist] Using lxml.etree.fromstring to parse untrusted XML data is known to be vulnerable to XML attacks. Replace lxml.etree.fromstring with its defusedxml equivalent function.
       Severity: Medium   Confidence: High
       Location: eemeter/io/parsers.py:465
    464	            if isinstance(xml, six.string_types + (six.binary_type,)):
    465	                self.root = etree.fromstring(xml)  # xml is a string.
    466	        self.timezone = self._get_timezone()

----

Happily the fix is easy. We just swap with the wrapper library, `defusedxml`. For more info:

* https://pypi.python.org/pypi/defusedxml
* http://lxml.de/FAQ.html#how-do-i-use-lxml-safely-as-a-web-service-endpoint

ℹ️ Note that it doesn't seem like a likely attack vector for the CLI,
and may not even be an attack vector in eemeter's SaaS offering.

However becuase it's straightforward to change it, seems worth doing.

---------------------------------------------------------

## 🤷‍♂️  Just _accept_ bandit's security scan finding re: FTP (no mitigation available without a complex change) (2d9cb98)

In short we just do not have another option for now. Added some notes about doing some authentication of these endpoints later on.

### CODE REVIEW

**Comment by philngo** [comment]
Thanks for these suggestions, Bandit looks very useful! We are removing dependency on FTP and XML in the 2.0.0 release.
